src/trans.rs

```diff
@@ -331,41 +353,165 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                 self.fun_types.insert(self.sig.clone(), ty);
             }
 
+            // Create the function prologue
+            // TODO: the epilogue and prologue are not always necessary
+            let read_sp = BinaryenLoad(self.module, 4, 0, 0, 0, BinaryenInt32(),
+                                       BinaryenConst(self.module, BinaryenLiteralInt32(0)));
```

Comment:
There should be a constant somewhere with the address of the stack pointer. This will help with documentation and also make it easier if we ever need to change the stack pointer.